enumerable.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    to_a.length.times { |i| yield self[i] }
+    self
+  end
+
+  def my_each_with_index
+    to_a.my_each { |each| yield each, index(each) }
+    self
+  end
+
+  def my_select
+    array = []
+    to_a.my_each { |each| array.push(each) if yield each }
+    array
+  end
+
+  def my_all?(parameter = nil)
+    to_a.my_each do |each|
+      if block_given?
+        return false unless yield each
+      elsif parameter.instance_of? Class
+        return false unless each.is_a? parameter
+      else
+        return false unless each
+        return false unless parameter === each
+      end
+    end
+    true
+  end
```

Comment:
## my_all
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
```ruby
true_array.my_all?
```
expected true
             got false